habitat_baselines/common/tensor_dict.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+
+import copy
+import numbers
+from typing import Callable, Dict, Optional, Tuple, Union, overload
+
+import numpy as np
+import torch
+
+TensorLike = Union[torch.Tensor, np.ndarray, numbers.Real]
+DictTree = Dict[str, Union[TensorLike, "DictTree"]]
+TensorIndexType = Union[int, slice, Tuple[Union[int, slice], ...]]
+
+
+class TensorDict(Dict[str, Union["TensorDict", torch.Tensor]]):
```

Comment:
Can we add some unit test for the class to cover main usage scenarios. Thank you!

Reply:
Yep